.github/workflows/ci.yml

```diff
@@ -0,0 +1,24 @@
+name: Avocado Gating CI
+
+on: [push, pull_request]
```

Comment:
I took a quick look at the documentation of `on`. Does this make sense to run the actions for all branches?

For Avocado, I think it may make sense to run the actions for `pull_request` and limited branches using the `push` option with specific arguments, like `master`. What do you think?

Reply:
I need to double check, but afaict, limiting to master will not run the workflow on your personal branch.

Reply:
Okay, I see. Today we don't run the CI on personal fork branches unless we manually enable it. With this approach, we are enabling CI on users' fork anyway. This will consume their CI hours and, maybe, that is not desired.

That is why I mentioned restricting the `push` to `master` only. But I think this is a matter of experimenting.

Reply:
This is a fair point, @willianrampazzo.

However, we are seeing an increase in minutes usage on our main repo, reducing the load on the main repo but first looking for problems on personal repos could be useful.

But yes, I understand your point, can we remove this in the future if we found any problem with that approach?

Reply:
As I mentioned, let's experiment and see how it behaves. We can change it later if we notice it is harmful.

Reply:
@willianrampazzo Thanks for the pointer! Indeed, I was caught by surprise when I was the CI running for our fork considering our usual expectation is that the main product is the one to run the CI before merges. Does this really consume GH action hours for everyone? Is there a way for some of us that would test once they establish a PR (i.e. an explicit candidate for testing for contributors) to disable this?